### PR TITLE
[NP-3054] Make UCJ status updates more restrictive

### PR DESCRIPTION
**Pull request by KernelDeimos**

### CODE REVIEW

**Comment by jlhughes** [comment]
@KernelDeimos Regarding capabilities with grant mode manual, other than system context, user will need this permission correct? 
Also with regard to test cases where we are calling updateJunction - the status is simply a requested status? 
@nanoscott tagging you so you are aware, thinking of API calls and the API user needing the new perm.

**Comment by KernelDeimos** [comment]
Yes that's correct, this permission is needed for other users to grant manual capabilities. For `updateJunction` the status can be considered requested status - the use case is that a MinMax will set a choice to `AVAILABLE` as a means of cancelling the UCJ if the user changes their mind

File: src/foam/nanos/crunch/ruler/AuthorizeUCJStatusOnPut.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    'foam.nanos.auth.AuthService',
+    'foam.nanos.crunch.Capability',
+    'foam.nanos.crunch.UserCapabilityJunction',
+    'java.util.HashMap',
+    'static foam.nanos.crunch.CapabilityJunctionStatus.*'
+  ],
+
+  documentation: `
+    This rule prevents a UCJ status from being set above ACTION_REQUIRED if the
+    user in context does not have
```
Comment (kgrgreer) [inline src/foam/nanos/crunch/ruler/AuthorizeUCJStatusOnPut.js]: Do you need to finish this sentence?

File: src/foam/nanos/crunch/ruler/AuthorizeUCJStatusOnPut.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            var ol = (UserCapabilityJunction) oldObj;
+            var nu = (UserCapabilityJunction) obj;
+            var auth = (AuthService) x.get("auth");
+
+            if ( ol == null ) {
+              if ( ! auth.check(x, PERMISSION) ) nu.setStatus(ACTION_REQUIRED);
+              return;
+            }
+
+            if (
```
Comment (kgrgreer) [inline src/foam/nanos/crunch/ruler/AuthorizeUCJStatusOnPut.js]: Why format this 'if' funny?

File: src/foam/nanos/crunch/ruler/AuthorizeUCJStatusOnPut.js (hunk 1 above)
Comment (KernelDeimos) [reply src/foam/nanos/crunch/ruler/AuthorizeUCJStatusOnPut.js]: Weird... it happened again. I think I have an editor shortcut I'm not aware of that does this :/

File: src/foam/nanos/crunch/ruler/AuthorizeUCJStatusOnPut.js (hunk 2 above)
Comment (KernelDeimos) [reply src/foam/nanos/crunch/ruler/AuthorizeUCJStatusOnPut.js]: It was just going past 80 characters so it was making me scroll sideways - should I move that to one line?

**Comment by KernelDeimos** [comment]
Added DNM tag because I'm getting a new error on here when I merge with dev+master that doesn't happen on dev+master itself

**Comment by KernelDeimos** [comment]
Fixed it!

**Review by jlhughes — APPROVED** [review]

**Pull request merged**